### PR TITLE
Add hidden inputs for readonly entries; fixes #12156

### LINKED ISSUE #12156: Dropdown::show() does not create hidden elements for read-only fields
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.2

### Bug description

Dropdown::show() does not create hidden elements for read-only fields:
https://github.com/glpi-project/glpi/blob/7e0a2407051d8cd24953847eb439ce7846524f65/src/Dropdown.php#L180-L184

But for example Dropdown::showFromArray() does:
https://github.com/glpi-project/glpi/blob/7e0a2407051d8cd24953847eb439ce7846524f65/src/Dropdown.php#L2071-L2080

Is there are reason for this inconsistency? 

It causes some issues in case field is supposed to be read-only and required at the same time as it is not sent to the form..

### Relevant log output

_No response_

### Page URL

_No response_

### Steps To reproduce

_No response_

### Your GLPI setup information

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #12156

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

File: src/Dropdown.php [hunk 1]
```diff
@@ -177,10 +177,15 @@ public static function show($itemtype, $options = [])
         }
 
         if ($params['readonly']) {
-            return '<span class="form-control" readonly'
+            $output = '';
+            foreach ($names as $name) {
+                $output .= "<input type='hidden' name='".$param ['name']."' value='$name'>";
+            }
```
Comment (cedric-anne) [inline src/Dropdown.php]: I may be wrong, but the expected value is probably `$params['value']` (or `$params['values']` when `$params['multiple']` is true).
Also, a `[]` should probably be added to `$param['name']` when `$params['multiple']` is true.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/Dropdown.php (hunk 1 above)
Comment (trasher) [reply src/Dropdown.php]: You're right for `[]` in the name.

`$params['values']` are computed into `$names` a few lines before, but maybe is that only relevant for display? I have no idea, I have aslo no idea how to reproduce the original issue.

Comment (trasher) [reply src/Dropdown.php]: I've change both, `$names` is probably only computed for display.

File: src/Dropdown.php [hunk 2]
```diff
@@ -177,10 +177,20 @@ public static function show($itemtype, $options = [])
...
 
         if ($params['readonly']) {
-            return '<span class="form-control" readonly'
+            $output = '';
+            if ($params["multiple"]) {
+                $field_name = $params['name'] . "[]";
+            } else {
+                $field_name = $params['name'];
+            }
+            foreach ($params['values'] as $value) {
```
Comment (cedric-anne) [inline src/Dropdown.php:186]:
```suggestion
                $field_name = $params['name'] . "[]";
                $values = $params['values'];
            } else {
                $field_name = $params['name'];
                $values = [$params['value']];
            }
            foreach ($values as $value) {
```

**Comment by cedric-anne** [comment]
I am not sure it will not have unexpected side effect, and reporter do not even says what usecase would be fixed by this. I would prefer to push this fix on main branch, even if it means doing a backport on 10.0/bugfixes when we have more elements to test/validate on a real case

**Comment by trasher** [comment]
ping @orthagh ?

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**